timeseries/src/autogluon/timeseries/predictor.py

```diff
@@ -162,7 +167,7 @@ def __init__(
             if std_freq != str(self.freq):
                 logger.info(f"Frequency '{self.freq}' stored as '{std_freq}'")
             self.freq = std_freq
-        if eval_metric == "mean_wQuantileLoss":
+        if isinstance(eval_metric, str) and eval_metric == "mean_wQuantileLoss":
             # We don't use warnings.warn since DeprecationWarning may be silenced by the Python warning filters
```

Comment:
unrelated nit: I just saw this. isn't this what the users want in this case? to silence deprecation warnings?

Reply:
This is related to the strange behavior of Python that suppresses the `DeprecationWarnings` **by default** in most cases (#3465, https://peps.python.org/pep-0565/). I've added a TODO to revisit this before the release, potentially change to DeprecationWarning and make sure that it's raised correctly in all contexts.